src/utils/__tests__/getFlowType-test.js

```diff
@@ -232,6 +232,23 @@ describe('getFlowType', () => {
     }, raw: '{ a: string, b: ?xyz }'});
   });
 
+  it('handles ObjectTypeSpreadProperty', () => {
+    var typePath = statement(`
+      var x: { ...OtherProps, a: string };
+    `).get('declarations', 0).get('id').get('typeAnnotation').get('typeAnnotation');
+
+    expect(getFlowType(typePath)).toEqual({
+      name: 'signature',
+      type: 'object',
+      signature: {
+        properties: [
+          { key: 'OtherProps', value: { name: 'OtherProps', required: true } },
```

Comment:
I think it is not really clear what this is just from the json alone. It could be also interpreted as 
```js
type a = { OtherProps: OtherProps };
```
What i would try to do is resolve `OtherProps` and if it resolves to an other ObjectType within the same file then I would just inline this in the result. We do the same when resolving stuff here https://github.com/reactjs/react-docgen/blob/master/src/utils/resolveObjectKeysToArray.js#L59..L67

In the case when it cannot be resolved I'm not sure what the best approach is. We could either ignore the spread completely then, which we currently do for regular non-type ObjectSpread. Or somehow add a special field in the output aka "inherits" (or some better name) which just contains the names of the Type that was spreaded into the object. If I think about it spreading types is just some kind of multi inheritance and it is probably interesting to know. 

```js
{
  name: 'signature',
  type: 'object',
  includes/inherits/implements/??: [ 'OtherProps' ],
  signature: {
    properties: [
      { key: 'a', value: { name: 'string', required: true } },
    ]
  }
}
```

Not sure if that makes sense though.

Reply:
The logic for resolving the local types is already being done. You can see it in action in the component snapshot test, prop `OtherLocalProps`.

Yeah, the real question is what to do when the type is imported from another file, 90% of our use case internally. We definitely don't want to ignore the spread. I think for us at least, saying what the types are is pretty explanatory. It's typically things like `...AccessibilityProps`. 

An explicit inherits field is interesting. I am not familiar enough with this project and how we consume the output to know what is ideal here. I think I defer to you.

Reply:
This is an example of a component where we type spread props in and how we'd probably expect them to show up in docs. https://facebook.github.io/react-native/docs/touchableopacity.html#props